test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9261,13 +9261,10 @@ def create_vnodes(self, name=None, attrib=None, num=1, mom=None,
                 self.manager(MGR_CMD_CREATE, NODE, m_attr, natvnode)
         # only expect if vnodes were added rather than the nat vnode modified
         if expect and num > 0:
-            attrs = {}
-            attrs.update({'state': 'free'})
+            attrs = {'state': 'free'}
             attrs.update(attrib)
             for vn in new_vnodelist:
                 rv = self.expect(VNODE, attrs, id=vn)
-                if not rv:
-                    break
         else:
             rv = True
         return rv
```

Comment:
I don't believe rv can be anything but True.  expect() is going to either raise an exception or return True.  This means you'll either set it to True via the expect() or set it to True on the else side of your if.  I'm not sure what you want to do about this.